mesonbuild/coredata.py

```diff
@@ -141,34 +141,25 @@ class UserArrayOption(UserOption):
     def __init__(self, name, description, value, shlex_split=False, **kwargs):
         super().__init__(name, description, kwargs.get('choices', []), yielding=kwargs.get('yielding', None))
         self.shlex_split = shlex_split
-        self.value = self.validate_value(value, user_input=False)
-
-    def validate_value(self, value, user_input=True):
-        # User input is for options defined on the command line (via -D
-        # options). Users can put their input in as a comma separated
-        # string, but for defining options in meson_options.txt the format
-        # should match that of a combo
-        if not user_input:
-            if isinstance(value, str):
-                if not value.startswith('['):
-                    raise MesonException('Valuestring does not define an array: ' + value)
-                newvalue = ast.literal_eval(value)
-            else:
-                newvalue = value
-        else:
```

Comment:
After reading the commit list, it's not clear to me what the replacement is for parsing of options in cross files.

Reply:
Yes. In Meson files we always use explicit types as much possible. String splitting is only for data that comes from the outside because there is really no way to do it otherwise.

Reply:
ok, kept a small cleanup patch there, but not changing the behaviour.